### PR TITLE
Mturk showmessages

### PR DESCRIPTION
**Pull request by marcelhollerbach**
- gives the worker a real answer in the statistik.
- remove the useless submit button - I hope we always end with finish, otherwise we are having a ugly problem with mturk.

### CODE REVIEW

**Comment by LeanderK** [comment]
what happens when it doesn't end with finish (should not be happening)

**Comment by marcelhollerbach** [comment]
@LeanderK we would get problems with the worker, since we are forced to give them the possibility to submit the hit, also if this does not happen they wont get money, since there is no assignment we could accept.

File: src/main/resources/mturk/worker-ui/mturk.js [hunk 1]
```diff
@@ -30,11 +54,16 @@ function initMturk(platformName, workerServiceUrl, experiment) {
...
     });
+
     WorkerUI.onSubmitAnswer(function (viewData, submittedData) {
-        //TODO log this
+        console.log("Save answer "+submittedData.answer);
+        creativeAnswers.push(submittedData.answer)
     });
-    WorkerUI.onSubmitCalibration(function (data) {
-        //TODO log this
+    WorkerUI.onSubmitRating(function (data) {
```
Comment (simonkorz) [inline src/main/resources/mturk/worker-ui/mturk.js]: should be `function (viewData, submittedData)`. I know it's the same in the sample, but i updated that ;)
viewData will return data used to display the view, submittedData will be what the worker-service gets posted

File: src/main/resources/mturk/worker-ui/mturk.js [hunk 2]
```diff
@@ -30,11 +54,16 @@ function initMturk(platformName, workerServiceUrl, experiment) {
...
+        for (var rating in data) {
+            console.log("Save Rating "+rating.rating+" - "+rating.answer);
```
Comment (simonkorz) [inline src/main/resources/mturk/worker-ui/mturk.js]: I'm afraid you forgot about the key iterator thing here.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/mturk/MturkPlatform.java [hunk 3]
```diff
@@ -102,6 +140,18 @@ public Boolean isCalibrationAllowed() {
         return new UnpublishHIT(connection, id);
     }
 
+    public <A> List<A> getFullList(int startIndex, Function<Integer, CompletableFuture<List<A>>> producer) throws ExecutionException, InterruptedException {
```
Comment (simonkorz) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/mturk/MturkPlatform.java]: Please add javadoc here

File: src/main/resources/mturk/worker-ui/mturk.js (hunk 2 above)
Comment (marcelhollerbach) [inline src/main/resources/mturk/worker-ui/mturk.js]: Ohm, no not in this case, i dont know why but it works here :/

File: src/main/resources/mturk/worker-ui/mturk.js [hunk 4]
```diff
@@ -30,11 +54,16 @@ function initMturk(platformName, workerServiceUrl, experiment) {
...
     });
+
     WorkerUI.onSubmitAnswer(function (viewData, submittedData) {
-        //TODO log this
+        console.log("Save answer "+submittedData.answer);
+        creativeAnswers.push(submittedData.answer)
     });
-    WorkerUI.onSubmitCalibration(function (data) {
-        //TODO log this
+    WorkerUI.onSubmitRating(function (function (viewData, submittedData)) {
```
Comment (simonkorz) [inline src/main/resources/mturk/worker-ui/mturk.js]: Nice double function ;)

Comment (marcelhollerbach) [inline src/main/resources/mturk/worker-ui/mturk.js]: FML

**Pull request merged**